File: leetcode/0075-sort-colors/solution.ts
function sortColors(nums: number[]): void {
  const count: number[] = [0, 0, 0];

  for (let i = 0; i < nums.length; i++) {
    count[nums[i]]++;
  }

  let idx = 0;
  for (let i = 0; i < 3; i++) {
    for (let j = 0; j < count[i]; j++) {
      nums[idx++] = i;
    }
  }
}

File: leetcode/0075-sort-colors/solution.cpp
void sortColors(vector<int>& nums) {
    int count[3];

    for (int i = 0; i < nums.size(); i++) {
        count[nums[i]]++;
    }

    int idx = 0;
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < count[i]; j++) {
            nums[idx++] = i;
        }
    }
}
